src/Model/Endpoint.php

```diff
@@ -451,66 +635,103 @@ public function resourceClass($name = null)
      *
      * @param null|string $method The inflection method to use
-     *
      * @return null|string
+     * @deprecated 2.0.0 Use setInflectionMethod() and getInflectionMethod() instead.
      */
     public function inflectionMethod($method = null)
     {
-        if ($method === null) {
-            return $this->_inflectionMethod;
+        if ($method !== null) {
+            return $this->setInflectionMethod($method);
         }
 
-        return $this->_inflectionMethod = $method;
+        return $this->getInflectionMethod();
+    }
+
+    /**
+     * Set a new inflection method
+     *
+     * @param string $method The name of the inflection method
+     * @return string
+     */
+    public function setInflectionMethod($method)
+    {
+        $this->_inflectionMethod = $method;
+
+        return $this->_inflectionMethod;
+    }
+
+    /**
+     * Get the inflection method
+     *
+     * @return string|null
+     */
+    public function getInflectionMethod()
+    {
+        return $this->_inflectionMethod;
     }
 
     /**
      * Returns an instance of the Webservice used
      *
      * @param \Muffin\Webservice\Webservice\WebserviceInterface|string|null $webservice The webservice to use
-     *
      * @return $this|\Muffin\Webservice\Webservice\WebserviceInterface
+     * @deprecated 2.0.0 Use setWebservice() and getWebservice() instead.
      */
     public function webservice($webservice = null)
     {
-        if ((is_string($webservice)) || ($this->_webservice === null)) {
-            if ($webservice === null) {
-                $webservice = $this->endpoint();
-            }
-
-            $connection = $this->connection();
-            if (!$connection) {
-                throw new UnexpectedDriverException(__('No connection has been defined for this endpoint'));
-            }
-
-            $this->_webservice = $connection->webservice($webservice);
-
-            return $this->_webservice;
+        if ($webservice !== null) {
+            return $this->setWebservice($webservice);
         }
+
         if ($webservice === null) {
-            return $this->_webservice;
+            return $this->getWebservice();
         }
 
         $this->_webservice = $webservice;
 
         return $this;
     }
 
+    /**
+     * Set the webservice instance to be used for this endpoint
+     *
+     * @param \Muffin\Webservice\Webservice\WebserviceInterface|string $webservice The webservice to use
+     * @return \Muffin\Webservice\Webservice\WebserviceInterface
+     * @throws \Muffin\Webservice\Exception\UnexpectedDriverException When no driver exists for the endpoint
+     */
+    public function setWebservice($webservice)
+    {
+        $connection = $this->getConnection();
+        if (!$connection) {
+            throw new UnexpectedDriverException(__('No driver has been defined for this endpoint'));
+        }
+
+        $this->_webservice = $connection->webservice($webservice);
+
+        return $this->getWebservice();
```

Comment:
Should return `$this` instead.

Reply:
I'm not sure that it can. [Due to this](https://github.com/davidyell/Webservice/blob/cake-version-upgrade/src/Model/Endpoint.php#L716). As the Endpoint sets the Webservice by calling the Connection, and the Connection proxies the call through to the driver. It would mean that `Connection::setWebservice()` would be returning an instance of the Driver and not an instance of the Webservice.